### PR TITLE
Re-reland Xcode backend refactor

### PR DESCRIPTION
**Pull request by dnfield**
This should be it this time - the devicelab tests all now pass.

FLUTTER_FRAMEWORK_DIR is still required for adding plugins - it should probably be removed, but that will end up being a breaking change that can be avoided for now.

My template updates were missing setting the SWIFT_VERSION for profile.  I've also updated the default Podfile template(s) to be aware of our new configuration scheme.  I don't think that's strictly necessary for existing projects.

/cc @xster

### CODE REVIEW

File: packages/flutter_tools/templates/cocoapods/Podfile-objc [hunk 1]
```diff
@@ -4,6 +4,12 @@
 # CocoaPods analytics sends network stats synchronously affecting flutter build latency.
 ENV['COCOAPODS_DISABLE_STATS'] = 'true'
 
+project 'Runner', {
```
Comment (xster) [inline packages/flutter_tools/templates/cocoapods/Podfile-objc]: If you change this, do you need to change all the existing materialized Podfiles too?

File: packages/flutter_tools/templates/app/ios-swift.tmpl/Runner.xcodeproj/project.pbxproj.tmpl [hunk 2]
```diff
@@ -325,6 +325,7 @@
 				);
 				PRODUCT_BUNDLE_IDENTIFIER = {{iosIdentifier}};
 				PRODUCT_NAME = "$(TARGET_NAME)";
+				SWIFT_VERSION = 4.0;
```
Comment (xster) [inline packages/flutter_tools/templates/app/ios-swift.tmpl/Runner.xcodeproj/project.pbxproj.tmpl]: Is this related?

**Comment by xster** [comment]
LGTM

File: packages/flutter_tools/templates/app/ios-swift.tmpl/Runner.xcodeproj/project.pbxproj.tmpl (hunk 2 above)
Comment (dnfield) [reply packages/flutter_tools/templates/app/ios-swift.tmpl/Runner.xcodeproj/project.pbxproj.tmpl]: This was missing from the profile build configuration, and broke a swift related test that ran profile

File: packages/flutter_tools/templates/cocoapods/Podfile-objc (hunk 1 above)
Comment (dnfield) [reply packages/flutter_tools/templates/cocoapods/Podfile-objc]: Only if they adopt a profile configuration, and only to avoid a warning from cocoapods as far as I can tell.
